java/spring/containerWithTests/Application/sampleWebApp.FunctionalTests/src/main/java/functional_tests/SampleFunctionalTest.java

```diff
@@ -43,12 +47,11 @@ public void testAssertTitleWithSelenium() throws InterruptedException, Assertion
             }
             catch(AssertionError e)
             {
-                if(i == (numRetries - 1))
-                {
+                if(currentTimestamp > endTimestamp) {

```

Comment:
Wouldn't it be better to use the above condition here. You can calculate currentTimestamp here